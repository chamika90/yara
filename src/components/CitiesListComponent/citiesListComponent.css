.cities-container {
  width: 100%;
  justify-content: center;
  align-items: center;
}
.cities {
    width: 100%;
    background-color: rgb(162, 161, 160);
    display: flex;
    flex-wrap: wrap;
    padding: 20;
    justify-content: center;
  }

  .city {
    padding: 10px 10px;
    border-color: black;
    border-width: 1;
    border-style: solid;
    margin-top: 5px;
    margin-right: 5px;
  }
  .city-selected {
    background-color: dodgerblue;
    color: white;
  }