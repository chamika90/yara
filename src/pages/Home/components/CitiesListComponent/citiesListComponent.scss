@import '../../../../resources/colors.scss';

.cities-container {
  width: 100%;
  justify-content: center;
  align-items: center;
}

.cities {
  width: 100%;
  background-color: $gray3;
  display: flex;
  flex-wrap: wrap;
  padding: 20;
  justify-content: center;
}

.city {
  padding: 10px 10px;
  border-color: $black;
  border-width: 1;
  border-style: solid;
  margin-top: 5px;
  margin-right: 5px;
}

.city-selected {
  background-color: $blue2;
  color: $white;
}

.loader-container {
  display: flex;
  width: 50%;
  height: 50%; 
  justify-content: center;
  align-items: center;
  position:absolute
}