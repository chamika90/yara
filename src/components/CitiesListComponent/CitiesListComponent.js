import React, {useEffect, useState} from "react";
import useFetch from "../../hooks/useFetch";
import './citiesListComponent.css';
import {getCitiesListEndpoint} from '../../helpers/utils';

const CitiesListContainer = ({onSelectCity}) => {
    const {data, isLoading, error, fetchData} = useFetch();
    const [citiesList, setCitiesList] = useState([]);
    const [selectedCity, setSelectedCity] = useState();
    
    useEffect(() => {
        fetchData(getCitiesListEndpoint());
    },[])

    useEffect(() => {
        if(data) {
            setCitiesList(data.results);
        }
    },[data]);

    const handleCitySelect = (city) => {
        onSelectCity(city)
        setSelectedCity(city.city);
    }

    return  <div className="cities-container">
                <div className="cities" >
                    {citiesList.map((city) => <button className={`city ${city.city === selectedCity ? 'city-selected' : ''}`} 
                    key={city.city} onClick={() => {handleCitySelect(city)}}>{city.city}</button>)}
                </div>
            </div>
}

export default CitiesListContainer;