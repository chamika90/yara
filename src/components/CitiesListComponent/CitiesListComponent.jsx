import React, {useEffect, useState} from "react";
import useFetch from "../../hooks/useFetch";
import {getCitiesListEndpoint} from '../../helpers/utils';
import LoadingSpinner from "../LoadingSpinner/LoadingSpinner";
import ErrorComponent from "../ErrorComponent/ErrorComponent";
import './citiesListComponent.scss';

const CitiesListContainer = ({onSelectCity}) => {
    const {data, isLoading, error, fetchData} = useFetch();
    const [citiesList, setCitiesList] = useState([]);
    const [selectedCity, setSelectedCity] = useState();
    
    useEffect(() => {
        fetchData(getCitiesListEndpoint());
    },[])

    useEffect(() => {
        if(data) {
            setCitiesList(data.results);
        }
    },[data]);

    const handleCitySelect = (city) => {
        onSelectCity(city)
        setSelectedCity(city.city);
    }

    return  <div className="cities-container">
                <div className="cities" >
                    {citiesList.map((city) => <button className={`city ${city.city === selectedCity ? 'city-selected' : ''}`} 
                    key={city.city} onClick={() => {handleCitySelect(city)}}>{city.city}</button>)}
                </div>
                {isLoading && <div className="loader-container"> <LoadingSpinner /> </div>}
                {error && <ErrorComponent message={'Something went wrong. Please try again later'} />}
            </div>
}

export default CitiesListContainer;