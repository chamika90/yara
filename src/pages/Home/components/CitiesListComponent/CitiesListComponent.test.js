import { render } from "@testing-library/react";

import CitiesListComponent from "./CitiesListComponent";

describe('render cities list', () => {
    it('Should match snapshot', () => {
        const {toJSON}  = render(<CitiesListComponent />);
        expect(toJSON).toMatchSnapshot();
    });
});