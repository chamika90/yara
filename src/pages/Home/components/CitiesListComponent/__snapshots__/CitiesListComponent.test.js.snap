// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render cities list Should match snapshot 1`] = `undefined`;
